algosdk/future/transaction.py

```diff
@@ -600,12 +636,15 @@ def _undictify(d):
         votefst = d["votefst"]
         votelst = d["votelst"]
         votekd = d["votekd"]
+        sprfID = d["sprfkey"]
```

Comment:
Similar thing here. If you attempted to parse an old keyreg txn before state proof keys were introduced, this would raise a KeyError.